packages/TorneloScoresheet/src/components/Toolbar/ArbiterModeNavigation.tsx

```diff
@@ -33,84 +70,135 @@ const ArbiterModeNavigation: React.FC = () => {
+    viewHistoryTransition: (() => void) | undefined | null,
+    willLoseData: boolean,
+  ): Option[] => {
+    const voidReturn = () => {
+      return;
+    };
+    return [
+      makeNavigationOption(
+        'Select active event',
+        ICON_CLOCK,
+        activeEventTransition === null ? voidReturn : activeEventTransition,
+        activeEventTransition === null ? true : false,
+        willLoseData,
+      ),
+      makeNavigationOption(
+        'Assign new game',
+        ICON_CLOCK,
+        assignGameTransition === null ? voidReturn : assignGameTransition,
+        assignGameTransition === null ? true : false,
+        willLoseData,
+      ),
+      makeNavigationOption(
+        'Go to active game',
+        ICON_CLOCK,
+        voidReturn,
+        false,
+        willLoseData,
+      ),
+      makeNavigationOption(
+        'View game history',
+        ICON_CLOCK,
+        viewHistoryTransition === null ? voidReturn : viewHistoryTransition,
+        viewHistoryTransition === null ? true : false,
+        willLoseData,
+      ),
+    ];
+  };
   const makeNavigationOption = (
     name: string,
+    icon: React.FC<SvgProps>,
     transition: (() => void) | undefined,
+    disabled?: boolean,
+    notifyLossOfData: boolean = false,
   ): Option => {
     return {
       text: name,
-      onPress: () => {
+      icon,
+      onPress: async () => {
         setShowNavigationSheet(false);
+        if (notifyLossOfData) {
+          const confirmExit = await confirmLossOfData();
+          if (!confirmExit) {
+            return;
+          }
+        }
         if (transition) {
           transition();
         }
       },
       style: {
-        width: 300,
+        width: 350,
+        marginTop: 20,
+        height: 70,
       },
+      disabled,
+      iconHeight: 40,
     };
   };
   const arbiterNavigationOptions: Record<AppMode, Option[]> = {
     [AppMode.ResultDisplay]: [],
     [AppMode.TablePairing]: [],
     [AppMode.Recording]: [],
-    [AppMode.EnterPgn]: [
-      makeNavigationOption('Past Games', useEnterPgnState()?.viewPastGames),
-    ],
-
-    [AppMode.ArbiterRecording]: [
-      makeNavigationOption(
-        'Enter Pgn',
+    [AppMode.EnterPgn]:
+      makeNavigationOptions(
+        null,
+        useEnterPgnState()?.goToPairingSelection,
+        useEnterPgnState()?.viewPastGames,
+        false,
+      ) ?? [],
+    [AppMode.ArbiterRecording]:
+      makeNavigationOptions(
         useArbiterRecordingState()?.[1].goBackToEnterPgn,
-      ),
-      makeNavigationOption(
-        'Pairing Selection',
         useArbiterRecordingState()?.[1].goBackToPairingSelection,
-      ),
-      makeNavigationOption(
-        'Table Pairing',
-        useArbiterRecordingState()?.[1].goBackToTablePairing,
-      ),
-    ],
-    [AppMode.ArbiterTablePairing]: [
-      makeNavigationOption(
-        'Enter Pgn',
+        useArbiterRecordingState()?.[1].goToGameHistory,
```

Comment:
Just a general comment about using `[1]` in the code. Without knowing the architecture of the scoresheet app it is difficult to review why you are getting the second variable in a list and what it is?

I am just assuming what you have done here is correct.

Maybe it would be better to use a local variable

```
const [appState,appNav] = useArbiterTablePairingState() ?? []
```

Then it would be far easier to review the logic.